src/v/compat/run.cc

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "compat/run.h"
+
+#include "compat/check.h"
+#include "compat/metadata_dissemination_compat.h"
+#include "compat/raft_compat.h"
+#include "json/document.h"
+#include "json/prettywriter.h"
+#include "json/writer.h"
+#include "seastarx.h"
+#include "utils/base64.h"
+#include "utils/file_io.h"
+
+#include <seastar/core/thread.hh>
+
+namespace compat {
+
+template<typename... Types>
+struct type_list {};
+
+using compat_checks = type_list<
+  raft::timeout_now_request,
+  raft::timeout_now_reply,
+  raft::transfer_leadership_request,
+  raft::transfer_leadership_reply,
+  cluster::update_leadership_request>;
+
+struct compat_error final : public std::runtime_error {
+public:
+    explicit compat_error(std::string_view msg)
```

Comment:
I was thinking the constructor would take as parameters maybe name and version, then callers wouldn't have to construct a string error message, and instead this could have overridden `what()` to do that

Reply:
fixed (i think)